### PR TITLE
gRPC storage plugin docs

### LINKED ISSUE #2664: Add docs for Experimental gRPC store
Add configuration details needed for the cortex config file and steps to integrate the third-party database for index & chunk store.

### PR DESCRIPTION
**Pull request by VineethReddy02**
**What this PR does**: Docs for gRPC storage plugin. 

**Which issue(s) this PR fixes**:
Fixes #2664 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/guides/grpc-storage-plugin.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+slug: grpc-based-plugin
+---
+
+_This feature is currently experimental and is only supported for Chunks storage._
+
+A gRPC based server, which will implement the methods required by the chunk-store. All the configuration required by the storage back-end will be configured in the cortex configuration file or using the flags, leaving just the gRPC server end-point as the only config required by the gRPC Clients that live in the Cortex Chunk store package. This allows the user to run the storage system as a sidecar container or as an independent service also to scale according to their needs.
+
+### How it works
+
+In the cortex configuration file, add ```store``` and ```object_store``` as ```grpc-store``` and configure storage with grpc server address I.e address to the gRPC server which implements the cortex chunk store methods.
```
Comment (pracucci) [inline docs/guides/grpc-storage-plugin.md:14]:
```suggestion
In the cortex configuration file, add `store` and `object_store` as `grpc-store` and configure storage with plugin server endpoint (ie. the address to the gRPC server which implements the cortex chunk store methods).
```

File: docs/guides/grpc-storage-plugin.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  grpc_store: 
+    # gRPC server address
+    server_address: localhost:6666
+```
+
+
+For reference these are the gRPC based chunk store implementation for Mongo and Mysql:
+
+1. [gRPC based Cortex chunk store for Mongo](https://github.com/VineethReddy02/cortex-mongo-store)
+2. [gRPC based Cortex chunk store for Mysql](https://github.com/VineethReddy02/cortex-mongo-store)
```
Comment (pracucci) [inline docs/guides/grpc-storage-plugin.md:41]: Links to Mongo, while should link to MySQL

File: docs/guides/grpc-storage-plugin.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      period: 168h
+
+storage:
+  grpc_store: 
+    # gRPC server address
+    server_address: localhost:6666
+```
+
+
+For reference these are the gRPC based chunk store implementation for Mongo and Mysql:
```
Comment (pracucci) [inline docs/guides/grpc-storage-plugin.md:38]: I would move this to a dedicated paragraph `## Community plugins`

File: docs/guides/grpc-storage-plugin.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+title: "gRPC storage plugin"
+linkTitle: "gRPC storage plugin"
+weight: 3
+slug: grpc-based-plugin
+---
+
+_This feature is currently experimental and is only supported for Chunks storage._
+
+A gRPC based server, which will implement the methods required by the chunk-store. All the configuration required by the storage back-end will be configured in the cortex configuration file or using the flags, leaving just the gRPC server end-point as the only config required by the gRPC Clients that live in the Cortex Chunk store package. This allows the user to run the storage system as a sidecar container or as an independent service also to scale according to their needs.
```
Comment (pracucci) [inline docs/guides/grpc-storage-plugin.md:10]: I rephrased a bit, adding an introduction. What do you think?

```suggestion
Cortex chunks storage supports a **gRPC-based plugin system** to use alternative backends for the index and chunks store.

A store plugin is a gRPC-based server which implements the methods required by the index and chunks store. Cortex chunks storage schema is then configured to use the plugin as backend system and gRPC will be used to communicate between Cortex and the plugin.

For example, if you're deploying your Cortex cluster on Kubernetes, the plugin would run as a sidecar container of your Cortex pods and the Cortex's `-grpc-store.server-address` should be configured to the endpoint exposed by the sidecar plugin (eg. `localhost:<port>`).
```

**Review by pracucci — COMMENTED** [review]
Thanks @VineethReddy02 for taking the time to work on it! The overall page LGTM. I left few comments/suggestions and then I think we're good to go 🚀

File: docs/guides/grpc-storage-plugin.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+---
+title: "gRPC storage plugin"
+linkTitle: "gRPC storage plugin"
+weight: 3
+slug: grpc-based-plugin
+---
+
+_This feature is currently experimental and is only supported for Chunks storage._
+
+Cortex chunks storage supports a **gRPC-based plugin system** to use an alternative backends for the index and chunks store.
```
Comment (pracucci) [inline docs/guides/grpc-storage-plugin.md:10]:
```suggestion
Cortex chunks storage supports a **gRPC-based plugin system** to use alternative backends for the index and chunks store.
```

File: docs/guides/grpc-storage-plugin.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+storage:
+  grpc_store:
+    # gRPC server address
+    server_address: localhost:6666
+```
+
+## Community plugins
+
+
+For reference these are the gRPC based chunk store implementation for Mongo and Mysql:
```
Comment (pracucci) [inline docs/guides/grpc-storage-plugin.md:41]:
```suggestion
The following list shows Cortex storage plugins built and shared by the community:
```

**Review by pracucci — COMMENTED** [review]
Thanks @VineethReddy02 for iterating on my feedback! I left a couple of last suggestions, which I'm just going to apply so we can merge it, if you have nothing against it. Thanks again!

**Review by pstibrany — APPROVED** [review]

**Comment by VineethReddy02** [comment]
Thanks @pracucci for all the support throughout this feature. I definitely had a great time working with the cortex team.

**Pull request merged**